src/model/CoilSystemCoolingDXHeatExchangerAssisted.cpp

```diff
@@ -252,6 +254,24 @@ namespace model {
     setHeatExchanger(heatExchanger);
   }
 
+  CoilSystemCoolingDXHeatExchangerAssisted::CoilSystemCoolingDXHeatExchangerAssisted(const Model& model, const AirToAirComponent& heatExchanger)
+    : StraightComponent(CoilSystemCoolingDXHeatExchangerAssisted::iddObjectType(), model) {
+    OS_ASSERT(getImpl<detail::CoilSystemCoolingDXHeatExchangerAssisted_Impl>());
+
+    CoilCoolingDXSingleSpeed coolingCoil(model);
+    setCoolingCoil(coolingCoil);
+
+    if (heatExchanger.optionalCast<HeatExchangerAirToAirSensibleAndLatent>()) {
+      // no-op
+    } else if (heatExchanger.optionalCast<HeatExchangerDesiccantBalancedFlow>()) {
+      // no-op
+    } else {
+      LOG_AND_THROW("AirToAirComponent type '" << heatExchanger.briefDescription()
+                                               << "' not currently supported as a heat exchanger type for CoilSystemCoolingDXHeatExchangerAssisted.");
+    }
+    setHeatExchanger(heatExchanger);
+  }
```

Comment:
When you deal with Ctors that could throw and where you instantiate other objects, you need to `remove()` stuff if it does. Here I think sure your coolingCoil would end up orphaned?